File: src/com/nerdscorner/mvp/mvp/BaseComponent.java
package com.nerdscorner.mvp.mvp;

public abstract class BaseComponent {

    public static final String LANGUAGE_PLACEHOLDER = "LANGUAGE";
    public static final String LANGUAGE_EXTENSION_PLACEHOLDER = "LANGUAGE_EXTENSION";

    public static final String JAVA_FOLDER = "java";
    public static final String KOTLIN_FOLDER = "kotlin";
    public static final String JAVA_EXTENSION = "java";
    public static final String KOTLIN_EXTENSION = "kt";

    public abstract void rollback();
}

File: src/com/nerdscorner/mvp/mvp/BaseComponent.kt
package com.nerdscorner.mvp.mvp

abstract class BaseComponent {

    abstract fun rollback()

    companion object {
        const val LANGUAGE_PLACEHOLDER = "LANGUAGE"
        const val LANGUAGE_EXTENSION_PLACEHOLDER = "LANGUAGE_EXTENSION"

        const val JAVA_FOLDER = "java"
        const val KOTLIN_FOLDER = "kotlin"
        const val JAVA_EXTENSION = "java"
        const val KOTLIN_EXTENSION = "kt"
    }
}
